src/algebra/category/Mon/limits.lean

```diff
@@ -0,0 +1,176 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import algebra.category.Mon.basic
+import category_theory.limits.types
+import category_theory.limits.creates
+
+/-!
+# The category of (commutative) (additive) monoids has all limits
+
+Further, these limits are preserved by the forgetful functor --- that is,
+the underlying types are just the limits in the category of types.
+
+-/
+
+open category_theory
+open category_theory.limits
+
+universe u
+
+namespace Mon
+
+variables {J : Type u} [small_category J]
+
+-- FIXME: to_additive by default transports this to `AddMon.monoid_obj`
+@[to_additive AddMon.add_monoid_obj]
+instance monoid_obj (F : J ⥤ Mon) (j) :
+  monoid ((F ⋙ forget Mon).obj j) :=
+by { change monoid (F.obj j), apply_instance }
```

Comment:
@cipher1024, could I get some help on this? Perhaps I'm just using `to_additive` incorrectly. I was hoping that I could just write `@[to_additive]` here, and have it automatically transfer the `Mon.monoid_obj` declaration to `AddMon.add_monoid_obj`. Instead, it transfers it to `AddMon.monoid_obj`.